project/Jdk9.scala

```diff
@@ -11,36 +11,87 @@
  * Copyright (C) 2017-2022 Lightbend Inc. <https://www.lightbend.com>
  */
 
-import sbt._
 import sbt.Keys._
+import sbt._
 
 object Jdk9 extends AutoPlugin {
   import JdkOptions.notOnJdk8
 
+  // The version 9 is special for any Java versions >= 9
+  // and the version 11 is special for any Java versions >= 11
+  // and the version 17 is special for any Java versions >= 17
+  // and the version 21 is special for any Java versions >= 21
+  private val supportedJavaLTSVersions = List("9", "11", "17", "21")
+
   lazy val CompileJdk9 = config("CompileJdk9").extend(Compile)
 
   lazy val TestJdk9 = config("TestJdk9").extend(Test).extend(CompileJdk9)
 
-  val SCALA_SOURCE_DIRECTORY = "scala-jdk-9"
-  val SCALA_TEST_SOURCE_DIRECTORY = "scala-jdk9-only"
-  val JAVA_SOURCE_DIRECTORY = "java-jdk-9"
-  val JAVA_TEST_SOURCE_DIRECTORY = "java-jdk9-only"
+  lazy val SCALA_SOURCE_DIRECTORY = getAdditionalSourceDirectoryNames("scala", isTest = false)
+  lazy val SCALA_TEST_SOURCE_DIRECTORY = getAdditionalSourceDirectoryNames("scala", isTest = true)
+
+  lazy val JAVA_SOURCE_DIRECTORY = getAdditionalSourceDirectoryNames("java", isTest = false)
```

Comment:
the name need to change to JAVA_SOURCE_DIRECTORIES or better still, use Scala naming convention of JavaSourceDirectories. JAVA_TEST_SOURCE_DIRECTORY should be renamed too.

Reply:
Change it directly, I just back to my home, very bad network.

Reply:
This is not my PR. It is also not even urgent. It is not normal practice to request other people to edit your PR for you. You can change this PR when you have free time. If that free time is in a week's time, then no harm.

Reply:
I just done it locally, will try to update...GFW.